plugins/docker/docker_stats_input.go

```diff
@@ -0,0 +1,229 @@
+/***** BEGIN LICENSE BLOCK *****
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# The Initial Developer of the Original Code is the Mozilla Foundation.
+# Portions created by the Initial Developer are Copyright (C) 2014
+# the Initial Developer. All Rights Reserved.
+#
+# Contributor(s):
+#   Anton Lindström (carlantonlindstrom@gmail.com)
+#   Rob Miller (rmiller@mozilla.com)
+#   Karl Matthias (karl.matthias@gonitro.com)
+#   Guy Templeton (guy.templeton@skyscanner.net)
+#
+# ***** END LICENSE BLOCK *****/
+
+package docker
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"math"
+	"github.com/mozilla-services/heka/pipeline"
+	"github.com/mozilla-services/heka/message"
+	"github.com/pborman/uuid"
+)
+
+type DockerStatsInputConfig struct {
+	// A Docker endpoint.
+	Endpoint      		string   `toml:"endpoint"`
+	CertPath      		string   `toml:"cert_path"`
+	NameFromEnv   		string   `toml:"name_from_env_var"`
+	FieldsFromEnv 		[]string `toml:"fields_from_env"`
+	FieldsFromLabels 	[]string `toml:"fields_from_labels"`
+}
+
+type DockerStatsInput struct {
+	conf         *DockerStatsInputConfig
+	stopChan     chan error
+	closer       chan struct{}
+	statsstream  chan *DockerStat
+	attachErrors chan error
+	statsMgr     *StatsManager
+	pack	     *pipeline.PipelinePack
+}
+
+func (di *DockerStatsInput) ConfigStruct() interface{} {
```

Comment:
This should use the same connect code as the other two plugins here. There's a method in `attach_manager.go` that does what you want.